### ISSUE TITLE
Lua branch Neovim crash with matchit.vim plugin

### ISSUE THREAD

**Issue by bluz71** [lukas-reineke/indent-blankline.nvim#79]
Hello,

I think indent-blankline is triggering an upstream Neovim crash.

Minimal `init.vim`:

```viml
set mouse=a
set termguicolors

call plug#begin('~/.vim/plugged')
Plug 'lukas-reineke/indent-blankline.nvim', {'branch': 'lua'}
call plug#end()

runtime macros/matchit.vim
```

Source Ruby file to edit: [ruby.rb](https://github.com/bluz71/vim-nightfly-guicolors/blob/master/samples/ruby.rb)

Mouse click to line 28 and do:

- `o` then enter the following text `def index2<CR><Esc>` (`<CR>` being `Enter` key)
- `u` then to immediately undo

Tip, typing the insertion text and undo quickly is more likely to trigger the crash. I am finding about 1 in 5 runs causes a crash as seen in this screenshot:

![Crash](https://user-images.githubusercontent.com/11382509/113246512-dd7c5f80-9304-11eb-96da-1e7d561eef08.png)

Note: sometimes I get: `ui_compositor.c:565 Assertion` crash, other times I get `tui.c:1390: tui_raw_line: Assertion`.

This happens with up-to-date Neovim and indent-blankline.

Cheers.

**Comment by lukas-reineke** [comment]
I can't reproduce it.
But this is clearly a Neovim bug anyways. There is not much I can do.
Can you try to reproduce it without indent-blankline by just setting virtual text yourself?

This would set virtual text on line 28, same as the plugin does.
```
:lua ns_id = vim.api.nvim_create_namespace("foo")
:lua vim.api.nvim_buf_set_extmark(0, ns_id, 28, 0, { virt_text={{'hello world', 'String'}}, virt_text_pos='overlay'})
```

**Comment by bluz71** [comment]
> I can't reproduce it.

That's a shame. I forgot to note, I am running on Linux.
Which platform are you on?

Note, I have found that quickly opening the file, clicking to line 28, doing the edits and then undoing has a better chance to crash. Quicker the better. If it doesn't crash the first time, then exit Neovim and come back in.

> Can you try to reproduce it without indent-blankline by just setting virtual text yourself?

I can't make it crash that way unfortunately. I have tried.
